### PR TITLE
Add multiprocessing

### LINKED ISSUE #17: Add ability to distribute components over multiple nodes in a cluster.
Right now all of the components are run as threads in a single process.  That is going to cause us to run into memory problems in the long run.  Add a system for managing distributed calculations.

### PR DESCRIPTION
**Pull request by rplzzz**
Allow components to be distributed over several nodes in a cluster.  Communication between nodes is handled via MPI.  Therefore, to use Cassandra in this mode you must have MPI and mpi4py installed.

Fixes #17

### CODE REVIEW

**Comment by rplzzz** [comment]
I have verified that this works on the component configuration in `example.cfg`.  I think I need to do a bit more testing, just to be sure that it works under a variety of conditions, but I think we're close enough to the final form that we can get started on the code review.

File: cassandra/rab.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
+"""Definition of the RAB class.
+
+The RAB provides access to capabilities hosted on remote nodes in MP
+calculations.  It implements much of the component interface, but it isn't a
+component as such, and in some respects it works very differently to proper
+components.
+
+The name RAB was originally an acronym for "Remote Access Broker", but mostly
+it's just a name.  Also, "RAB" are my mom's initials.  Hi, Mom!
```
Comment (calebbraun) [inline cassandra/rab.py]: 😂

File: cassandra/rab.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+import threading
+import logging
+
+# The choice of the sleep length in RAB listener loop is a tradeoff.  Making it
+# shorter improves responsiveness, but creates more waiting in the GIL,
+# potentially slowing down threads that are actually running models (especially
+# if those models are implemented in python and running in the same process).  
+RAB_LOOP_SLEEP = 0.05          # 50 ms
+
+class RAB(object):
```
Comment (calebbraun) [inline cassandra/rab.py]: Inheriting from `object` is the default in python 3, so this could just be `class RAB:`.

Comment (rplzzz) [reply cassandra/rab.py]: Good to know, thanks.

File: cassandra/rab.py (hunk 1 above)
Comment (rplzzz) [reply cassandra/rab.py]: She said she felt "honored" when I told her.

**Pull request merged**